pkg/client/acr/acr.go

```diff
@@ -90,27 +98,47 @@ func (c *Client) Tags(ctx context.Context, host, repo, image string) ([]api.Imag
 			host, err)
 	}
 
-	var tags []api.ImageTag
+	// Create a map of tags, so that when we come up with additional Tags
+	// we can add them as Children
+	tags := map[string]api.ImageTag{}
+
 	for _, manifest := range manifestResp.Manifests {
-		if len(manifest.Tags) == 0 {
-			tags = append(tags, api.ImageTag{
-				SHA:       manifest.Digest,
-				Timestamp: manifest.CreatedTime,
-			})
+		// Base data shared across tags
+		base := api.ImageTag{
+			SHA:          manifest.Digest,
+			Timestamp:    manifest.CreatedTime,
+			OS:           manifest.OS,
+			Architecture: manifest.Architecture,
+		}
 
+		// No tags, use digest as the key
+		if len(manifest.Tags) == 0 {
+			tags[base.SHA] = base
 			continue
 		}
 
 		for _, tag := range manifest.Tags {
-			tags = append(tags, api.ImageTag{
-				SHA:       manifest.Digest,
-				Timestamp: manifest.CreatedTime,
-				Tag:       tag,
-			})
+			current := base   // copy the base
+			current.Tag = tag // set tag value
+
+			// Already exists — add as child
+			if parent, exists := tags[tag]; exists {
+				parent.Children = append(parent.Children, &current)
+				tags[tag] = parent
+			} else {
+				// First occurrence — assign as root
+				tags[tag] = current
+			}
 		}
 	}
 
-	return tags, nil
+	// Convert Map to Slice
+	taglist := make([]api.ImageTag, 0, len(tags))
+	for _, tag := range tags {
+		taglist = append(taglist, tag)
+	}
+
+	return taglist, nil
```

Comment:
acr, ecr and selfhosted all have the same logic for building a taglist from tags. Could this be reused in `util.go`?

Reply:
Thats fair! 👍  - I didn't know how much or many times I'd use it 🙈.